### PR TITLE
HW-53998: Fee section should not be displayed if there's no fee

### PR DESCRIPTION
**Pull request by oleksii-lubianyi-epam**

### CODE REVIEW

File: Sources/Receipt/ReceiptDetailSectionData.swift [hunk 1]
```diff
@@ -113,10 +113,13 @@ struct ReceiptDetailSectionFeeData: ReceiptDetailSectionData {
                                      field: "amount"))
         var fee: Double = 0.0
         if let strFee = receipt.fee {
```
Comment (jsaini-hw) [inline Sources/Receipt/ReceiptDetailSectionData.swift]: change strFee to stringFee

File: Sources/Receipt/ReceiptDetailSectionData.swift [hunk 2]
```diff
@@ -113,10 +113,13 @@ struct ReceiptDetailSectionFeeData: ReceiptDetailSectionData {
...
-            rows.append(ReceiptDetailRow(title: "receipt_details_fee".localized(),
-                                         value: String(format: valueCurrencyFormat, strFee, receipt.currency),
-                                         field: "fee"))
             fee = Double(strFee) ?? 0.0
+
+            if fee != 0.0 {
```
Comment (jsaini-hw) [inline Sources/Receipt/ReceiptDetailSectionData.swift]: The ticket is to not display the entire section if there's no fee. So add an if check in Presenter:
`if let receiptFeeSection = ReceiptDetailSectionFeeData(from: receipt) {
            sectionData.append(receiptFeeSection)
        }`

and update ReceiptDetailSectionFeeData to initialize only if there's fee else return nil.

File: Sources/Receipt/ReceiptDetailSectionData.swift (hunk 1 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Receipt/ReceiptDetailSectionData.swift]: Changed

File: Sources/Receipt/ReceiptDetailSectionData.swift (hunk 2 above)
Comment (oleksii-lubianyi-epam) [reply Sources/Receipt/ReceiptDetailSectionData.swift]: Thanks, updated

File: Tests/Receipt/ReceiptDetailPresenterTests.swift [hunk 3]
```diff
@@ -23,6 +23,16 @@ class ReceiptDetailPresenterTests: XCTestCase {
         XCTAssertEqual(presenterWithNotes.sectionData.count, 4)
     }
 
+    func testSectionFeeIsNilIfNoOrZeroFee() {
```
Comment (jsaini-hw) [inline Tests/Receipt/ReceiptDetailPresenterTests.swift]: rename it to testSectionFeeIsNil

File: UITests/Receipts/PrepaidCardListReceiptTests.swift [hunk 4]
```diff
@@ -133,13 +145,6 @@ class PrepaidCardListReceiptTests: BaseTests {
         XCTAssertEqual(transactionDetails.receiptIdValue.label, "FISVL_5269000")
         // Will comment out for now
         //XCTAssertEqual(transactionDetails.dateValue.label, "Thu, Jun 20, 2019, 9:23 PM")
-
-        // Fee
-        XCTAssertEqual(transactionDetails.feeSection.label, "Fee Specification")
```
Comment (jsaini-hw) [inline UITests/Receipts/PrepaidCardListReceiptTests.swift]: Add test case to verify fee section

File: Tests/Receipt/ReceiptDetailPresenterTests.swift (hunk 3 above)
Comment (oleksii-lubianyi-epam) [reply Tests/Receipt/ReceiptDetailPresenterTests.swift]: Renamed

File: UITests/Receipts/PrepaidCardListReceiptTests.swift (hunk 4 above)
Comment (oleksii-lubianyi-epam) [reply UITests/Receipts/PrepaidCardListReceiptTests.swift]: Added verification that fee section doesn't exist.

File: UITests/Receipts/PrepaidCardListReceiptTests.swift [hunk 5]
```diff
@@ -145,13 +145,7 @@ class PrepaidCardListReceiptTests: BaseTests {
...
-        XCTAssertEqual(transactionDetails.amountLabel.label, "Amount:")
-        XCTAssertEqual(transactionDetails.transactionLabel.label, "Transaction:")
-        XCTAssertEqual(transactionDetails.amountValue.label, "10.00 USD")
-        // XCTAssertEqual(transactionDetails.transactionValue.label, "10.00 USD")
+        XCTAssertTrue(!transactionDetails.feeSection.exists)
```
Comment (jsaini-hw) [inline UITests/Receipts/PrepaidCardListReceiptTests.swift]: XCTAssertFalse

File: UITests/Receipts/PrepaidCardListReceiptTests.swift (hunk 4 above)
Comment (jsaini-hw) [reply UITests/Receipts/PrepaidCardListReceiptTests.swift]: The above comment was to add 1 UI Test to validate Fee Section exists. That UI tests still needs to be added

File: UITests/Receipts/PrepaidCardListReceiptTests.swift (hunk 5 above)
Comment (oleksii-lubianyi-epam) [reply UITests/Receipts/PrepaidCardListReceiptTests.swift]: Fixed

File: UITests/Receipts/PrepaidCardListReceiptTests.swift (hunk 4 above)
Comment (oleksii-lubianyi-epam) [reply UITests/Receipts/PrepaidCardListReceiptTests.swift]: @jsaini-hw Sorry, I still cannot get your point. This test case (as well as the next one) supposes fee section not exists.  
Fee section exists and is being validated within `testPrepaidReceiptDetail_verifyTransactionOptionalFields()`.
Could you please provide me more details on expected test cases?

Comment (jsaini-hw) [reply UITests/Receipts/PrepaidCardListReceiptTests.swift]: @oleksii-lubianyi-epam as long as it is tested, I am good

**Pull request merged**